### PR TITLE
formulaic 0.6.2

### PR DESCRIPTION
**Pull request by skupr-anaconda**
Changelog: https://github.com/matthewwardrop/formulaic/releases
License: https://github.com/matthewwardrop/formulaic/blob/v0.6.2/LICENSE
Requirements: https://github.com/matthewwardrop/formulaic/blob/v0.6.2/pyproject.toml

Actions:
1. Apply an initial **percy** [patch](https://github.com/anaconda-distribution/percy/blob/main/percy/examples/patch/test_patch.json) and run locally a [script](https://github.com/anaconda-distribution/percy/blob/main/percy/examples/patch/updater_standalone.py) to: 
  - Remove `noarch python`
  - Skip `py<37`
  - Add flags `--no-deps --no-build-isolation` to `script`
  - Add missing `setuptools` and `wheel` to `host`
  - Fix `python` in `host` and `run`
  - Add `pip check`
2. Rename `hatch` as `hatchling`, see https://github.com/matthewwardrop/formulaic/blob/v0.6.2/LICENSE
3. Update runtime pinnings and add constraints

Notes:
- `lifelines` requires it https://github.com/conda-forge/lifelines-feedstock/blob/532bab4fccd8407f9e4e4010ab6733951ef254e8/recipe/meta.yaml#L37

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -10,26 +10,28 @@ source:
   sha256: 54905a297a40f6d8b23b94b6652d622ad4756a2b14fb83c0aa76473254d4dd7b
 
 build:
-  noarch: python
   number: 0
-  script: {{ PYTHON }} -m pip install . -vv
+  script: {{ PYTHON }} -m pip install . --no-deps --no-build-isolation --ignore-installed --no-cache-dir -vvv
+  skip: True  # [py<37]
```
Comment (M-Waszkiewicz-Anaconda) [inline recipe/meta.yaml:15]: what if someone has python 3.7.1 ?

Comment (skupr-anaconda) [reply recipe/meta.yaml:15]: Eh, yeah. I guess we need a `run_constrained` with `python >=3.7.2`

Comment (skupr-anaconda) [reply recipe/meta.yaml:15]: Done

**Review by M-Waszkiewicz-Anaconda — APPROVED** [review]
LGTM

**Review by varlackc — COMMENTED** [review]
We might need to remove the `abs.yaml` file before merging to defaults

**Review by akabanovs — APPROVED** [review]
LGTM

**Pull request merged**